### PR TITLE
Work in some changes to Master

### PR DESCRIPTION
**Pull request by beansrowning**
# Changes  

## Pre-processing, feature engineering  

- Current to origin/scott to c16efdfa9424167e6d4274d42a9f04f8e9afb90a  
- Improves feature_exraction.py using main() block to avoid threading issues  
- Improves tools.multi.get_times() by using context manager for Pool() to exit gracefully  
- features_to_integers performs all text vectorization
- First, last, or both cut points for data prior to model run performed in model_prep.py  
  - Correctly trims the prediction horizon appropriately  
  - Updated find_cutpoints() to handle sequence trimming on the left to MAX_LEN, or 0

## First Keras model
  - Two data generators: one using keras.utils.Sequence (Dense only), and one using tf.data (Ragged or Dense)
  - Starter keras model works with ragged tensors now (shape: (batch_size, None, None)), eliminating the need for padding along two dimensions and speeding up processing  
  - Hyperparameter tuning available via kerastuner in hp_tuning.py

## Misc  
- Linting support moved from Black -> Yapf  
- Updated requirements  
- Updated data pointer, now contains a few model runs (tensorboard + weights) in the private data repo  
- Probably other things

### CODE REVIEW

**Comment by beansrowning** [comment]
## Merging after our discussion today
